### PR TITLE
admin tasks

### PR DESCRIPTION
**Pull request by dstephens3**

### CODE REVIEW

**Review by harperiscoding — CHANGES_REQUESTED** [review]
fails travis ci

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+class TasksController < ApplicationController
+  before_action :set_task, only: [:destroy, :update, :show, :edit]
+  before_action :set_event
+
+  def new
+    @task = Task.new
+  end
+
+  def index
+    @task = Task.all_tasks(@event.id)
```
Comment (harperiscoding) [inline app/controllers/tasks_controller.rb]: since you have a relation there's an easier way of doing this.

File: config/routes.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
   resources :events do
     resources :guests, only: [:new, :create, :destroy]
-    resources :tasks
+    resources :tasks, only: [:new, :create, :destroy, :edit, :show, :update, :index]
```
Comment (harperiscoding) [inline config/routes.rb]: you're using only but you're including all restful routes

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      redirect_to new_event_task_url
+    end
+  end
+
+  def edit
+  end
+
+  def create
+    some = params["task"]
+    some["event_id"] = @event.id
```
Comment (harperiscoding) [inline app/controllers/tasks_controller.rb]: this whole "some" thing is super obtuse and hard to read, try to refactor using some built in methods.

File: app/models/task.rb [hunk 4]
```diff
@@ -3,12 +3,19 @@ class Task < ApplicationRecord
   has_many :volunteers, :through => :assignments
   has_one :event
 
+  validates :name, presence: true
+
+  def self.all_tasks(id)
```
Comment (harperiscoding) [inline app/models/task.rb]: there's a built in method to do this.

File: config/application.yml.asc [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-OxJ6wCMARRwA2mQbcWZ/NSSFwzNNaURaJ/voAVpgBU5IOqAGggze4cgucMCAOFbF
-wfZ03KHkIRxZmHoa1V+hnrL+stzpj2SnR2Dr2TjUd3Wh9cO5HaaWIYn8kX2RTQhA
-SA88QpV9jMhHAeyZnKqqF32M3BfkP6sK2bkg2Sa0dHIIGZAerUbdD+j06VFkvI+3
-ReFp9jrbORBIayhwYp10QVlfQsa/94+pZ/aBr4KQbwYcZ2xrOdKeDf81Notyo2Sj
-H/YUiep4WmdY3HqsP3wlVdCidtMPJUmsi0d2h9atv9X26i/L4NPIylzgLuu1s9WQ
-T/IZI402lxihMxJPQcARhX0hHtKHI8dowYudm1GbGokPng0i5HTratagHXa6Djfy
-epREic7jAzkeZAdkPXKBkPl0TQPSZdjpTwXi6v7UQkiXbnVm5s+f7GfnRPjLAhkE
-Fd5cXcik1fHn74YsHB0vNUg+BV8t4Yfi3h2eHVLYVgu2P+L8VfdPiZ9+G9GV0ZFV
-BllElTrRsiXQdy3MjAo0/lyxjEXuJ7mCPo9PF6gBQNX0GvG711YlYeBevJiBEBTK
-bqbUpcyx7VTLxzHdMZmYYmmhaV9LdR/Z6XgjONLNzVdBh6yqPN0OBDsWCA==
```
Comment (harperiscoding) [inline config/application.yml.asc]: not sure why this has changed.

File: features/tasks.feature [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      | name              | description                 | date        | location                               | tickets               | capacity |
+      | Pouring liquid    | Look, it changes colors     | 22-Apr-3019 | 422 Treeside Way, Berkeley, CA 94704   | stubhub.com/liquid    | 10       |
+    And I am on the events page
+    And I follow "Show"
+
+  Scenario: I add a task:
+    When I follow "Add Task"
+    When I fill out the form with the following attributes:
+      | name   | Test           |
+      | description     | a chore     |
```
Comment (harperiscoding) [inline features/tasks.feature]: condense this test into steps instead of using web steps.

**Review by harperiscoding — CHANGES_REQUESTED** [review]

File: app/controllers/tasks_controller.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+  def show
+  end
+
+  def edit
+  end
+
+  def create
+    @task = Task.new(task_params)
+    @task.event = @event
```
Comment (harperiscoding) [inline app/controllers/tasks_controller.rb:15]: this is already done for you in your code.

File: app/views/tasks/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<%= form_with(model: task, url: [event, task], local: true) do |form| %>
+
+
```
Comment (harperiscoding) [inline app/views/tasks/_form.html.erb]: spacing.

File: config/routes.rb [hunk 9]
```diff
@@ -3,6 +3,7 @@
     resources :guests, only: [:new, :create, :destroy]
     resources :tasks
     resources :assignments, except: [:show, :update]
+
```
Comment (harperiscoding) [inline config/routes.rb]: spacing.

File: features/tasks.feature [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+  Background:
+    Given the following events exist:
+      | name              | description                 | date        | location                               | tickets               | capacity |
+      | Pouring liquid    | Look, it changes colors     | 22-Apr-3019 | 422 Treeside Way, Berkeley, CA 94704   | stubhub.com/liquid    | 10       |
+    And for Pouring liquid the following tasks exist:
+      | name              | description                 | deadline        |
+      | get liquid   | Look, it changes colors     | 28-Apr-3019 |
+    And I am on the events page
+    And I follow "Show"
```
Comment (harperiscoding) [inline features/tasks.feature]: why two and I follow statements in a row? should be in a step.

**Review by harperiscoding — CHANGES_REQUESTED** [review]

File: features/support/paths.rb [hunk 11]
```diff
@@ -20,6 +20,7 @@ def path_to(page_name)
     when /^the new donation page$/ then '/donations/new'
     when /^the first task edit page$/ then '/events/1/tasks/1/edit'
     when /^the add task page$/ then '/events/1/tasks/new'
+    when /^the tasks page$/ then '/events/1/tasks'
```
Comment (harperiscoding) [inline features/support/paths.rb]: step does not say anything about it only being for the first event.

**Review by harperiscoding — CHANGES_REQUESTED** [review]
still haven't removed the unnecessary @events assignment, and there are unresolved merge conflicts.

File: features/support/paths.rb (hunk 11 above)
Comment (dstephens3) [reply features/support/paths.rb]: I tried to remove the @events assignment but then the task did not retain its event info correctly.

Comment (dstephens3) [reply features/support/paths.rb]: also i am fixing merge conflicts now and then i will repush

**Pull request merged**